CDCgov/prime-reportstream
# 16084: add support for submitting reports via blob trigger

This PR adds the ability to process reports via SFTP upload

Test Steps:
1. Open azure storage explorer
2. Create a new blob container `sftp-submissions`
3. In the new container, create the directory structure `ignore/ignore-full-elr`
4. Add an HL7 file to the new directory
5. Confirm that the file is processed by the pipeline

## Changes
- New blob trigger that handles files getting added to a storage container
- Steps for setting up SFTP ingestion

## Checklist

### Testing

### Process

## Linked Issues
- https://github.com/CDCgov/prime-reportstream/issues/16084

## To Be Done
- Devops ticket for setting up SFTP in azure
- Devops (nice to have future improvement): #16262

## Specific Security-related subjects a reviewer should pay specific attention to
